validphys2/src/validphys/results.py

```diff
@@ -799,8 +799,13 @@ def groups_chi2_table(groups_data, pdf, groups_chi2, groups_each_dataset_chi2):
     for group, groupres, dsresults in zip(groups_data, groups_chi2, groups_each_dataset_chi2):
         for dataset, dsres in zip(group, dsresults):
             stats = chi2_stats(dsres)
-            stats["group"] = dataset.name
+            stats["dataset"] = dataset.name
+            stats["group"] = str(group)
             records.append(stats)
+        stats = chi2_stats(groupres)
+        stats["group"] = str(group)
+        stats["dataset"] = ""
```

Comment:
Is this for total? I thought that was computed further down the line

Reply:
No, at this point the total (per group) is already computed.
There's another function (fits_chi2?) with the total computed separately though.

Reply:
```suggestion
        stats["dataset"] = "Total"
```
Okay I was worried this would have some weird effects later, but I think it's okay.

Reply:
I changed it to "Group Total"